### PR TITLE
Added metrics_collection management to ec2_asg module to switch ASG m…

### PR DESCRIPTION
**Pull request by stefanhorning**
…etrics on and off.

##### SUMMARY
Added metric_collection functionality to ec2_asg module.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
ec2_asg module

##### ANSIBLE VERSION
```
ansible 2.4.2.0
  config file = None
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Nov 20 2017, 18:23:56) [GCC 5.4.0 20160609]
```

##### ADDITIONAL INFORMATION

Used following boto3 method (for reference): http://boto3.readthedocs.io/en/latest/reference/services/autoscaling.html#AutoScaling.Client.enable_metrics_collection

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/ec2_asg.py [hunk 1]
```diff
@@ -173,6 +173,34 @@
...
+    required: False
+    default: False
+    choices: ['True', 'False']
+    version_added: "2.5"
+  metrics_granularity:
+    description:
+      - When metrics_collection is enabled this will determine granularity of metrics collected by CloudWatch
+    required: False
+    default: "1minute"
+    choices: ['1minute']
```
Comment (resmo) [inline lib/ansible/modules/cloud/amazon/ec2_asg.py]: according code, there is no choice list

File: lib/ansible/modules/cloud/amazon/ec2_asg.py [hunk 2]
```diff
@@ -173,6 +173,34 @@
...
+    description:
+      - Enable ASG metrics collection
+    required: False
+    default: False
+    choices: ['True', 'False']
+    version_added: "2.5"
+  metrics_granularity:
+    description:
+      - When metrics_collection is enabled this will determine granularity of metrics collected by CloudWatch
+    required: False
```
Comment (resmo) [inline lib/ansible/modules/cloud/amazon/ec2_asg.py]: required=False is the default, can be omitted

File: lib/ansible/modules/cloud/amazon/ec2_asg.py [hunk 3]
```diff
@@ -173,6 +173,34 @@
     default: []
     choices: ['Launch', 'Terminate', 'HealthCheck', 'ReplaceUnhealthy', 'AZRebalance', 'AlarmNotification', 'ScheduledActions', 'AddToLoadBalancer']
     version_added: "2.3"
+  metrics_collection:
+    description:
+      - Enable ASG metrics collection
+    required: False
```
Comment (resmo) [inline lib/ansible/modules/cloud/amazon/ec2_asg.py]: required=False is the default, can be omitted

File: lib/ansible/modules/cloud/amazon/ec2_asg.py [hunk 4]
```diff
@@ -173,6 +173,34 @@
...
+    default: False
+    choices: ['True', 'False']
```
Comment (resmo) [inline lib/ansible/modules/cloud/amazon/ec2_asg.py]: better use `type: bool` instead of `choices: ['True', 'False']`

**Review by resmo — DISMISSED** [review]
some minor doc issues found

**Comment by stefanhorning** [comment]
Fixed all the docs issues, please revise.

**Comment by s-hertel** [comment]
@stefanhorning This looks good. Since ec2_asg is a stableinterface module and this is a new feature, [integration tests need to be added](https://github.com/ansible/community/blob/master/group-aws/integration.md#policy) to https://github.com/ansible/ansible/tree/devel/test/integration/targets/ec2_asg to ensure backwards compatibility.

**Review by s-hertel — APPROVED** [review]

**Comment by ryansb** [comment]
Looks good - I'll come back to merge this after freeze is lifted (as this isn't a community module)

**Review by ryansb — APPROVED** [review]

**Pull request merged**